### PR TITLE
setup.sh: Download pip packages inside virtualenv

### PR DESCRIPTION
**Pull request by Kamorst**
Do not litter the `$HOME` of the user installing pip packageds inside the home user directory. Instead download pip packages inside a virtualenv only when building the man pages and documentation.

### CODE REVIEW

File: setup.sh [hunk 1]
```diff
@@ -90,24 +91,46 @@ function check_dependencies()
...
 
+function generate_documentation()
+{
+  local ret
+
+  python3 -m venv "$DOCS_VENV"
+
+  source "${DOCS_VENV}/bin/activate"
+  install_pip_dependencies
+  ret=$?
```
Comment (rodrigosiqueira) [inline setup.sh:121]: Add double quotes around "$?".

File: setup.sh [hunk 2]
```diff
@@ -331,8 +354,18 @@ function synchronize_files()
 
   # man file
   mkdir -p "$mandir"
+
+  python3 -m venv "$DOCS_VENV"
+
+  source "${DOCS_VENV}/bin/activate"
+  install_pip_dependencies
+  ret=$?
```
Comment (rodrigosiqueira) [inline setup.sh:362]: Double-quotes.
Do you need it? If so, you must add a check for the return.

**Comment by siddhanthrathod** [comment]
This is great for operating systems as gentoo which don't allow pip to install packages system wide and --user flag or a --target is needed !

File: documentation/dependencies/arch.dependencies [hunk 3]
```diff
@@ -1,22 +1,18 @@
...
 graphviz
-python-virtualenv
 texlive-bin
 librsvg
 bzip2
 lzip
 lzop
 zstd
 xz
-python-pip
```
Comment (rodrigosiqueira) [inline documentation/dependencies/arch.dependencies:19]: Maybe I'm missing something, but if we remove `python-pip` how can you call `pip` in a fresh env that does not have `pip` installed by default?

File: setup.sh [hunk 4]
```diff
@@ -41,10 +41,11 @@ declare -r DOCUMENTATION='documentation'
 
 declare -r CONFIGS_PATH='configs'
 
+declare -r DOCS_VENV='docs_venv'
```
Comment (rodrigosiqueira) [inline setup.sh:44]: What the `V` in `VENV`  means? Can we have a more readable name?

File: setup.sh [hunk 5]
```diff
@@ -90,22 +91,32 @@ function check_dependencies()
...
+
+  python3 -m venv "$DOCS_VENV"
+
+  # Activate python virtual env
+  source "${DOCS_VENV}/bin/activate"
+  say 'Creating python virtual env...'
+  cmd="pip --quiet --require-virtualenv install --requirement \"${DOCUMENTATION}/dependencies/pip.dependencies\""
+  eval "$cmd"
+  ret="$?"
+  if [[ "$ret" == 0 ]]; then
```
Comment (rodrigosiqueira) [inline setup.sh:108]: Add a new line before the `if` condition.

File: setup.sh [hunk 6]
```diff
@@ -90,22 +91,32 @@ function check_dependencies()
...
+    complain 'Could not install pip dependencies'
+  fi
+  # Deactivate python virtual env
+  deactivate
 
-    # Install pip packages
-    cmd="pip install $pip_package_list"
-    eval "$cmd"
+  if [[ -d "$DOCS_VENV" ]]; then
+    rm -r "$DOCS_VENV"
```
Comment (rodrigosiqueira) [inline setup.sh:117]: Use `--recursive` instead of `-r`.

**Review by rodrigosiqueira — CHANGES_REQUESTED** [review]

File: documentation/dependencies/arch.dependencies (hunk 3 above)
Comment (Kamorst) [reply documentation/dependencies/arch.dependencies:19]: `python3 -m venv` will install pip inside the virtual env. So the host do not need to install it system-wide.

File: setup.sh (hunk 4 above)
Comment (Kamorst) [reply setup.sh:44]: `V` stands for "virtual". I will change it.

**Pull request closed without merging**